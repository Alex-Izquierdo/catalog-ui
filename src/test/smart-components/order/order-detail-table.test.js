import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

import OrderDetailTable from '../../../smart-components/order/order-detail-table';

describe('<OrderDetailTable />', () => {
  let initialProps;

  beforeEach(() => {
    initialProps = {
      requests: [{
        reason: 'foo',
        requester: 'bar',
        updated_at: 'Wed Apr 10 2019 14:26:42 GMT+0200 (Central European Summer Time)',
        state: 'baz'
      }]
    };
  });

  it('should render correctly', () => {
    const wrapper = shallow(<OrderDetailTable { ...initialProps } />);
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});
