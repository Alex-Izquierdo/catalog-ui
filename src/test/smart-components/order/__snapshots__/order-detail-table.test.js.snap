// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<OrderDetailTable /> should render correctly 1`] = `
<table
  className="requests-table"
>
  <thead>
    <tr>
      <th>
        Steps
      </th>
      <th>
        Performed by
      </th>
      <th>
        Date & time
      </th>
      <th>
        Status
      </th>
    </tr>
  </thead>
  <tbody>
    <tr
      className=""
      key="0"
    >
      <td>
        <StepLabel
          index={0}
          text="foo"
        />
      </td>
      <td>
        bar
      </td>
      <td>
        <span>
          Wed Apr 10 2019 14:26:42 GMT+0200 (Central European Summer Time)
        </span>
      </td>
      <td>
        baz
      </td>
    </tr>
  </tbody>
</table>
`;
